tests/test_pipeline_media.py

```diff
@@ -249,3 +257,61 @@ def test_use_media_to_download_result(self):
         self.assertEqual(new_item['results'], [(True, 'ITSME')])
         self.assertEqual(self.pipe._mockcalled, \
                 ['get_media_requests', 'media_to_download', 'item_completed'])
+
+
+class MediaPipelineAllowRedirectSettingsTestCase(unittest.TestCase):
```

Comment:
@kmike , I added some resources to MockServer to list files and redirect, and used similar tests to `tests/test_crawl.py`